README.md

```diff
@@ -1,2 +1,462 @@
+        dns: www.example.com
+        ca: self-sign
+        key_usage:
+          - digitalSignature
+          - nonRepudiation
+          - keyEncipherment
+        extended_key_usage:
+          - id-kp-clientAuth
+          - id-kp-serverAuth
+  roles:
+    - certificate
+```
+
+
+### Don't wait for the certificate to be issued
+
+The certificate issuance can take several minutes depending on the CA.
+Because of that it's also possible to request the certificate but not
+actually wait for it.
+
+```yaml
+
+---
+- hosts: webserver
+  vars:
+    certificate_wait: false
+    certificate_requests:
+      - name: mycert
+        dns: www.example.com
+        ca: self-sign
+  roles:
+    - certificate
+```
+
+
+### Set a principal
+
+```yaml
+
+---
+- hosts: webserver
+  vars:
+    certificate_requests:
+      - name: mycert
+        dns: www.example.com
+        ca: self-sign
+        principal: HTTP/www.example.com@EXAMPLE.COM
+
+  roles:
+    - certificate
+```
+
+
+### Choosing to not auto-renew a certificate
+
+By default certificates generated by the role will be set for
+auto-renewal. To disable that behavior set `auto_renew: no`.
+
+```yaml
+
+---
+- hosts: webserver
+  vars:
+    certificate_requests:
+      - name: mycert
+        dns: www.example.com
+        ca: self-sign
+        auto_renew: no
+
+  roles:
+    - certificate
+```
+
+
+### Using FreeIPA to issue a certificate
+
+If your host is enrolled in a FreeIPA server you will also have the option
+to use it's CA to issue your certificate. To do that just set `ca: ipa`.
+
+```yaml
+
+---
+- hosts: webserver
+  vars:
+    certificate_requests:
+      - name: mycert
+        dns: www.example.com
+        principal: HTTP/www.example.com@EXAMPLE.COM
+        ca: ipa
+
+  roles:
+    - certificate
+```
+
+### Running a command before or after a certificate is issued
+
+
+Sometimes you need to execute a command just before a certificate is
+renewed and another command just after. In order to do that use
+`run_before` and `run_after`.
+
+```yaml
+
+---
+- hosts: webserver
+  vars:
+    certificate_requests:
+      - name: mycert
+        dns: www.example.com
+        ca: self-sign
+        run_before: systemctl stop webserver.service
+        run_after: systemctl start webserver.service
+
+  roles:
+    - certificate
+```
+
+### Setting the certificate owner and group
+
+If you are using a certificate for a service, for example httpd,
+it's likely that you'll need to set the certificate owner and group
+that will own the certificate. In the following example the
+owner and group are both set to httpd.
+
+
+```yaml
+
+---
+- hosts: webserver
+  vars:
+    certificate_requests:
+      - name: mycert
+        dns: www.example.com
+        ca: self-sign
+        owner: httpd
+        group: httpd
+
+  roles:
+    - certificate
+```
+
+Please note that it's also possible to use UID and GID instead of
+user and group names.
+
+
+## Compatibility
+
+Currently supports CentOS / RHEL 7+.
```

Comment:
Fedora?

Reply:
Debian/Ubuntu?

Reply:
At least Debian 10 was tested, I believe.

Reply:
Added a note for Debian 10.